docs/public/specifications/tests/Address-TP/address02.md

```diff
@@ -1,46 +1,102 @@
-## ADDRESS02: Reverse DNS entry exists for name server IP address
+# ADDRESS02: Reverse DNS entry exists for name server IP address
 
-### Test case identifier
-**ADDRESS02** Reverse DNS entry should exist for name server IP address
+## Test case identifier
+**ADDRESS02** 
 
-### Objective
+## Table of contents
 
-Some anti-spam techniques use reverse DNS lookup to allow incoming traffic.
-In order to prevent name servers to be blocked or blacklisted, DNS 
-administrators should publish PTR records associated to name server
-addresses.
+* [Objective](#objective)
+* [Scope](#scope)
+* [Inputs](#inputs)
+* [Summary](#summary)
+* [Test procedure](#test-procedure)
+* [Outcome(s)](#outcomes)
+* [Special procedural requirements](#special-procedural-requirements)
+* [Intercase dependencies](#intercase-dependencies)
 
-TODO: Technical reference to be found
+## Objective
 
-### Inputs
+Best curent practices dictates that imternet reachable hosts should have a
+reverse DNS entry, as various services on the Internet, for instance spam 
+filters, may consider this when determining the trustworthiness of the host.
+Se [RFC1912] section 2.1 and [RFC1033] page 11 for additional information.
+
+## Scope
+
+This test checks for the existence of PTR records for the corresponding reverse
+domains of the name servers IP address.
+
+## Inputs*
 
 The domain name to be tested.
 
-### Ordered description of steps to be taken to execute the test case
+## Summary
+
+Message Tag                   | Level    | Arguments | Message ID for message tag
+:---------------------------- |:---------|:----------|:--------------------------
+A02_PTR_PRESENT               | INFO     |           | PTR present for all nameservers
+A02_PTR_MISSING               | NOTICE   | ns_list   | PTR missing for "{ns_list}"
+
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
 
-1. Obtain the glue address records  of the domain checked
-   using [Method4](../Methods.md)
+## Test procedure 
 
-2. Obtain the IP addresses of each name server of the domain checked
-   using [Method5](../Methods.md)
+1. Create the empty set: Name server name and IP address ("Name Server IP").
 
-3. For each IP address, a recursive PTR query must be performed.
+2. Obtain the glue address records of each name server for the domain from the
+   parent using the method [Get-Del-NS-Names-and-IPs] and add them to the 
+   *Name Server IP* set. 
```

Comment:
With the method we get all name server names in the delegation and all IP addresses for those names, but the IP addresses do not have to be from glue records. They could be out-of-bailiwick and fetched seperately.

Compare with the update in https://github.com/zonemaster/zonemaster/pull/1284